phoonnx_train/optispeech/values.py

```
@@ -0,0 +1,169 @@
+class InferenceInputs(BaseValueContainer):
+    clean_text: str
+    x: IntArray
+    x_lengths: IntArray
+    sids: IntArray | None = None
+    lids: IntArray | None = None
+    d_factor: float = 1.0
+    p_factor: float = 1.0
+    e_factor: float = 1.0
+
+    @classmethod
+    def from_ids_and_lengths(cls, ids: list[int], lengths: list[int], **kwargs) -> "Self":
```

Ruff (0.15.1)

[error] Undefined name `Self`

(F821)